src/Compilers/CSharp/Test/Emit2/Diagnostics/DiagnosticAnalyzerTests.cs

```diff
@@ -4009,6 +4009,46 @@ public record A(int X, int Y);";
                      Diagnostic("MyDiagnostic", @"public record A(int X, int Y);").WithLocation(2, 1));
         }
 
+        [DiagnosticAnalyzer(LanguageNames.CSharp)]
+        public class PrimaryConstructorBaseTypeAnalyzer : DiagnosticAnalyzer
+        {
+            public const string DiagnosticId = "MyDiagnostic";
+            internal const string Title = "MyDiagnostic";
+            internal const string MessageFormat = "MyDiagnostic";
+            internal const string Category = "Category";
+
+            internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, Category, DiagnosticSeverity.Warning, isEnabledByDefault: true);
+
+            public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+            public override void Initialize(AnalysisContext context)
+            {
+                context.RegisterSyntaxNodeAction(AnalyzePrimaryConstructorBaseType, SyntaxKind.BaseList, SyntaxKind.PrimaryConstructorBaseType);
+            }
+
+            private static void AnalyzePrimaryConstructorBaseType(SyntaxNodeAnalysisContext context)
+            {
+                var diagnostic = CodeAnalysis.Diagnostic.Create(Rule, context.Node.GetLocation());
```

Comment:
>var diagnostic = CodeAnalysis.Diagnostic.Create(Rule, context.Node.GetLocation()); [](http://example.com/codeflow?start=16&length=82)

Consider also verifying symbol associated with the context.

Reply:
Done